src/digest/sha1.cr

```diff
@@ -10,23 +10,33 @@ class Digest::SHA1 < Digest::Base
   # This is a direct translation of https://tools.ietf.org/html/rfc3174#section-7
   # but we use loop unrolling for faster execution (about 1.07x slower than OpenSSL::SHA1).
 
+  @intermediate_hash = uninitialized UInt32[5]
+  @length_low = 0_u32
+  @length_high = 0_u32
+  @message_block_index = 0
+  @message_block = StaticArray(UInt8, 64).new(0_u8) # uninitialized UInt8[64]
+
   def initialize
-    @intermediate_hash = uninitialized UInt32[5]
+    reset
+  end
+
+  def reset : self
+    @message_block.each_index do |i|
```

Comment:
Assigning a new instance might be preferable?

Reply:
No.  See the benchmark results above.  Iterative hashing is one example where performance is critical.  Reducing the speed reduces the maximum security parameters.

Reply:
I mean `@message_block = StaticArray(UInt8, 64).new(0_u8)`, not initializing a new `Digest` instance. `reset` is fine.

Reply:
Oh.

Is there a `memset`?

Reply:
I removed it.  It's not used.